Branch-SDK/Branch-SDK/BranchViewHandler.m

```diff
@@ -0,0 +1,123 @@
+//
+//  BranchViewHandler.m
+//  Branch-TestBed
+//
+//  Created by Sojan P.R. on 3/3/16.
+//  Copyright © 2016 Branch Metrics. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+#import "BranchViewhandler.h"
+#import "Branch.h"
+#import "BranchView.h"
+
+@interface BranchViewHandler() <UIWebViewDelegate>
+@end
+
+NSString * const BRANCH_VIEW_REDIRECT_SCHEME = @"branch-cta";
+NSString * const BRANCH_VIEW_REDIRECT_ACTION_ACCEPT = @"accept";
+NSString * const BRANCH_VIEW_REDIRECT_ACTION_CANCEL = @"cancel";
+
+@implementation BranchViewHandler
+
+static BranchViewHandler *branchViewHandler;
+BOOL isBranchViewAccepted = NO;
+NSString *currentActionName;
+NSString *currentBranchViewID;
+
++ (BranchViewHandler *)getInstance {
+    if (!branchViewHandler) {
+        branchViewHandler = [[BranchViewHandler alloc] init];
+    }
+    return branchViewHandler;
+}
+
+- (BOOL)showBranchView:(NSString *)actionName withBranchViewDictionary:(NSDictionary*)branchViewDict andWithDelegate:(id)callback {
```

Comment:
Good catch. We are  using it only to create the BranchView. Removed it passing to functions below.